### PR TITLE
Nex 2 monorepo

### PR DESCRIPTION
**Pull request by ekondareva**
*Link to ticket: [NEX-2 monorepo](https://wunder.atlassian.net/jira/software/c/projects/NEX/boards/389?modal=detail&selectedIssue=NEX-2)*

*Changes proposed in this PR:*
- Drupal related files moved to /drupal folder
- Project renamed to next4drupal-project

*How to test:*
1. Check no files are missing and no secret files are commited
2. Check lando.yml is correct (renaming of the project is successfully and fully done)
3. Locally checkout to the branch and run the following commands. Check that site is installed.

- lando start
- composer install
- import db
- drush cmi
- drush cr
- lando drush ucrt wunder with email and pwd (check ucrt command syntax)
- lando drush user-add-role "next_js_admin" wunder
- go to /admin/config/services
https://next4drupal-project.lndo.site/admin/config/people/simple_oauth
Generate keys to ../oath (folder /oath should be under /drupal but above /web)
- Go to 
go to /admin/config/services/consumer and create wunder consumer with next js admin role
- copy consumer uuid to .env.local in /next
- try out with creating a drupal article and viewing it in drupal and previewing in nextjs.
For that, in /next run 'npm run dev' and go to check if localhost:3000 works ok and shows your new article.
then in drupal View your article and open for preview in next.js

*Link to feature environment:*
<add silta url here>

### CODE REVIEW

**Review by tharna — CHANGES_REQUESTED** [review]
For security we should also have `*.sql*` rule so nobody commits sql dumbs by accident.

These rules (under drupal/.gitignore) should be enough to skip all the composer generated additional files and a few others we don't want to commit.
```
# Ignore directories generated by Composer
/.editorconfig
/drush/Commands/
/vendor/
/web/_ping.php
/web/**/.gitignore
/web/core/
/web/*/contrib/
/web/libraries/
/web/sites/default/default.*
/web/sites/example.*
/web/**/README.txt

# Ignore local configuration
/web/sites/*/settings.local.php

# Ignore Drupal's file directory
web/sites/*/files/

# Ignore testing directories.
/web/sites/simpletest
/web/tmp

# Ignore oauth keys
/oauth
```

**Comment by vermario** [comment]
hi! I was able to start the project and see the article in the next side.

I pushed a small update to the .env.example file to reflect the new lando domain, and added placeholders for the auth settings.

**Comment by vermario** [comment]
@tharna  @ekondareva I have updated the gitignore setup: 

* added gitignore inside drupal
* added generic gitignore rules in the root of the project
* added a specific gitignore for the drupal/oauth folder, so that we can keep the folder but gitignore everything inside it. (coming from https://stackoverflow.com/questions/115983/how-do-i-add-an-empty-directory-to-a-git-repository)

**Comment by vermario** [comment]
question: I noticed the scaffolded files are not committed (the files that you get when you run composer install): 
image

I think we normally commit them in projects right?

**Comment by ekondareva** [comment]
> question: I noticed the scaffolded files are not committed (the files that you get when you run composer install): image
> 
> I think we normally commit them in projects right?

That was my question too. I was not sure and did not commit them. But I can do that if needed. Or you can Mario commit them if you have them at hand and you think it is OK to keep them in repo.

File: drupal/.gitignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/web/modules/contrib/
+/web/themes/.gitignore
+/web/themes/contrib/
+/web/profiles/.gitignore
+/web/profiles/contrib/
+/web/libraries/
+/web/sites/.gitignore
+/web/sites/default/.gitignore
+
+/.idea
```
Comment (tharna) [inline drupal/.gitignore:18]: .idea is already ignored in root .gitignore

File: .gitignore [hunk 2]
```diff
@@ -1,29 +1,10 @@
...
-
-# Ignore testing directories.
-/web/sites/simpletest
-/web/tmp
+# Ignore potential database dumps:
+*.sql.*
+*.sql
 
 # Ignore files generated by PhpStorm
 /.idea/
```
Comment (tharna) [inline .gitignore:10]: Remove leading /

File: drupal/.gitignore [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+/web/core/
+/web/modules/.gitignore
+/web/modules/contrib/
+/web/themes/.gitignore
+/web/themes/contrib/
+/web/profiles/.gitignore
+/web/profiles/contrib/
+/web/libraries/
+/web/sites/.gitignore
+/web/sites/default/.gitignore
```
Comment (tharna) [inline drupal/.gitignore:16]: Could use /web/**/.gitignore to shave off a few lines instead of listing all the particular directories separately

**Review by tharna — CHANGES_REQUESTED** [review]
Few more small changes. Also add the scaffold files to drupal/.gitignore (see my comment earlier for suggested rules)

File: drupal/.gitignore (hunk 3 above)
Comment (vermario) [reply drupal/.gitignore]: done in https://github.com/wunderio/next4drupal-project/pull/10/commits/a7db4595229392194f98077a01456768b90f09e8

File: drupal/.gitignore (hunk 1 above)
Comment (vermario) [reply drupal/.gitignore]: done in https://github.com/wunderio/next4drupal-project/pull/10/commits/9b36fbdd4fbb64fc73d22620eca857d4abd75adc

File: .gitignore (hunk 2 above)
Comment (vermario) [reply .gitignore]: Done in https://github.com/wunderio/next4drupal-project/pull/10/commits/9b36fbdd4fbb64fc73d22620eca857d4abd75adc

**Comment by vermario** [comment]
> add the scaffold files to drupal/.gitignore (see my comment earlier for suggested rules)
 
 @tharna :) maybe it's my november brain, but I don't see rules to ignore the scaffold files (they are spread a bit all over, in folders with also files that belong to the project... did I miss something?

**Comment by tharna** [comment]
> > add the scaffold files to drupal/.gitignore (see my comment earlier for suggested rules)
> 
> @tharna :) maybe it's my november brain, but I don't see rules to ignore the scaffold files (they are spread a bit all over, in folders with also files that belong to the project... did I miss something?

```
/drush/Commands/
/vendor/
/web/_ping.php
/web/**/.gitignore
/web/core/
/web/*/contrib/
/web/libraries/
/web/sites/default/default.*
/web/sites/example.*
/web/**/README.txt
```

**Comment by vermario** [comment]
> > > add the scaffold files to drupal/.gitignore (see my comment earlier for suggested rules)
> > 
> > 
> > @tharna :) maybe it's my november brain, but I don't see rules to ignore the scaffold files (they are spread a bit all over, in folders with also files that belong to the project... did I miss something?
> 
> ```
> /drush/Commands/
> /vendor/
> /web/_ping.php
> /web/**/.gitignore
> /web/core/
> /web/*/contrib/
> /web/libraries/
> /web/sites/default/default.*
> /web/sites/example.*
> /web/**/README.txt
> ```
> 
> Rest probably need to be defined individually

I see! done in https://github.com/wunderio/next4drupal-project/pull/10/commits/f3f43b31aa3779cfb428f2cc87a695c77d8e50f8

**Review by tharna — APPROVED** [review]

**Pull request merged**